src/dataworker/DataworkerUtils.ts

```diff
@@ -100,7 +100,8 @@ export async function blockRangesAreInvalidForSpokeClients(
       blockRanges,
       chainIdListForBundleEvaluationBlockNumbers
     );
-    assert(Object.keys(endBlockTimestamps).length === Object.values(spokePoolClients).filter(isDefined).length);
+    // There should be a spoke pool client instantiated for every bundle timestamp.
+    assert(Object.keys(endBlockTimestamps).every((chainId) => spokePoolClients[chainId] !== undefined));
```

Comment:
Thoughts on switching to a negated `some`? It would potentially halt the loop faster 
```suggestion
    assert(!Object.keys(endBlockTimestamps).some((chainId) => !isDefined(spokePoolClients[chainId])));
```